connectors/sources/network_drive.py

```diff
@@ -382,13 +395,20 @@ def find_matching_paths(self, advanced_rules):
 
     async def ping(self):
         """Verify the connection with Network Drive"""
+        if "RUNNING_FTEST" in os.environ:
+            return
```

Comment:
We definitely should not add this in our connectors. Any ftest code should live outside unless it's really impossible to work around this (e.g. when we're using 3rd-party library most of the time that we need to change URL for).

Can you remove this code and approach it differently?